src/utils/update/psc.to.beneficial.owner.type.mapper.ts

```diff
@@ -0,0 +1,172 @@
+  const principal_address = mapAddress(psc.address);
+
+  const result: BeneficialOwnerOther = {
+    id: psc.links?.self,
+    name: psc.name,
+    principal_address: principal_address,
+    service_address: service_address,
+    is_service_address_same_as_principal_address: isSameAddress(service_address, principal_address) ? yesNoResponse.Yes : yesNoResponse.No,
+    legal_form: psc.identification?.legalForm,
+    law_governed: psc.identification?.legalAuthority,
+    public_register_name: psc.identification?.placeRegistered,
+    registration_number: psc.identification?.registrationNumber,
+    is_on_register_in_country_formed_in: undefined,
+    start_date: undefined,
+    // is_on_sanctions_list: psc.isSanctioned === "true" ? yesNoResponse.Yes : yesNoResponse.No,
+  };
+  mapNatureOfControl(psc, result, false);
+  return result;
+
+};
+
+export const mapPscToBeneficialOwnerGov = (psc: CompanyPersonWithSignificantControl): BeneficialOwnerGov => {
+  const service_address = mapAddress(psc.address);
+  const principal_address = mapAddress(undefined);
+
+  const result: BeneficialOwnerGov = {
+    id: psc.links?.self,
+    name: psc.name,
+    principal_address: principal_address,
+    service_address: service_address,
+    is_service_address_same_as_principal_address: isSameAddress(service_address, principal_address) ? yesNoResponse.Yes : yesNoResponse.No,
+    legal_form: psc.identification?.legalForm,
+    law_governed: psc.identification?.legalAuthority,
+    start_date: undefined,
+    // is_on_sanctions_list: psc.isSanctioned === "true" ? yesNoResponse.Yes : yesNoResponse.No,
+  };
+  mapNatureOfControl(psc, result, true);
+  return result;
+};
+
+const mapNatureOfControl = (psc: CompanyPersonWithSignificantControl, beneficialOwner: BeneficialOwnerIndividual| BeneficialOwnerOther, isBeneficialGov: boolean) => {
+  psc.naturesOfControl?.forEach(natureType => {
+    const controlKind = natureTypeMap.get(natureType);
+    switch (controlKind) {
+        case 'BoNatureOfControl':
+          beneficialOwner.beneficial_owner_nature_of_control_types = NatureOfControlTypeMap.get(natureType) as unknown as NatureOfControlType[];
+          break;
+        case "NonLegalNatureOfControl":
+          beneficialOwner.non_legal_firm_members_nature_of_control_types = NatureOfControlTypeMap.get(natureType) as unknown as NatureOfControlType[];
+          break;
+        case "TrustNatureOfControl":
+          if (!isBeneficialGov){
+            beneficialOwner.trustees_nature_of_control_types = NatureOfControlTypeMap.get(natureType) as unknown as NatureOfControlType[];
+          }
+          break;
+        default:
+          throw new Error('INVALID NATURE OF CONTROL TYPE');
+    }
+  });
+};
+
+enum BoTypes {
+  BO_NATURE_OF_CONTROL = "BoNatureOfControl",
+  TRUST_NATURE_OF_CONTROL = "TrustNatureOfControl",
+  NON_LEGAL_NATURE_OF_CONTROL = "NonLegalNatureOfControl",
+}
+
+export enum natureOfControlMap {
```

Comment:
This enum is not a Map - call it natureOfControlTypes? cf. line 91

Reply:
fixed